pallets/automation-time/src/exchange.rs

```diff
@@ -0,0 +1,143 @@
+// This file is part of OAK Blockchain.
+
+// Copyright (C) 2022 OAK Network
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// ! Traits and default implementation for paying execution fees.
+use crate::Config;
+
+use codec::FullCodec;
+use sp_runtime::{
+	traits::{AtLeast32BitUnsigned, CheckedSub, MaybeSerializeDeserialize, Zero},
+	DispatchError,
+	TokenError::BelowMinimum,
+};
+use sp_std::{fmt::Debug, marker::PhantomData};
+
+use frame_support::traits::{
+	Currency, ExistenceRequirement, Imbalance, OnUnbalanced, WithdrawReasons,
+};
+
+type NegativeImbalanceOf<C, T> =
+	<C as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
+
+/// Handle withdrawing, refunding and depositing of transaction fees.
+pub trait NativeTokenExchange<T: Config> {
+	/// The underlying integer type in which fees are calculated.
+	type Balance: AtLeast32BitUnsigned
+		+ FullCodec
+		+ Copy
+		+ MaybeSerializeDeserialize
+		+ Debug
+		+ Default
+		+ scale_info::TypeInfo;
+	type LiquidityInfo: Default;
+
+	/// The minimum balance any single account may have. This is equivalent to the `Balances`
+	/// module's `ExistentialDeposit`.
+	fn minimum_balance() -> Self::Balance;
+
+	///Transfer some liquid free balance to another staker.
```

Comment:
staker?

Reply:
Ya i just copied the documentation from the Balance pallet. I was debating between making it clearer to me vs making this is a straight pass through